aria_data_loaders/aria_data_utils/spot_aria_p1_node.py

```diff
@@ -0,0 +1,591 @@
+import logging
+import os
+from typing import Any, Dict, List, Optional, Tuple
+
+import click
+import cv2
+import numpy as np
+import rospy
+import sophus as sp
+from aria_data_utils.conversions import (
+    generate_PoseStamped_a_T_b_from_spSE3,
+    generate_spSE3_a_T_b_from_PoseStamped,
+    generate_spSE3_a_T_b_from_TransformStamped,
+    generate_TransformStamped_a_T_b_from_spSE3,
+    xyt2sophus,
+)
+from aria_data_utils.detector_wrappers.april_tag_detector import AprilTagDetectorWrapper
+from aria_data_utils.detector_wrappers.object_detector import ObjectDetectorWrapper
+from aria_data_utils.image_utils import decorate_img_with_text
+from aria_data_utils.perception.april_tag_pose_estimator import AprilTagPoseEstimator
+from bosdyn.client.frame_helpers import get_a_tform_b
+from fairotag.scene import Scene
+from geometry_msgs.msg import Pose, PoseStamped
+from matplotlib import pyplot as plt
+from projectaria_tools.core import calibration, data_provider, mps
+from scipy.spatial.transform import Rotation as R
+from spot_rl.envs.skill_manager import SpotSkillManager
+from spot_rl.models.owlvit import OwlVit
+from spot_wrapper.spot import Spot, SpotCamIds, image_response_to_cv2
+from std_msgs.msg import Bool
+from tf2_ros import (
+    ConnectivityException,
+    ExtrapolationException,
+    LookupException,
+    StaticTransformBroadcaster,
+)
+from tf2_ros.buffer import Buffer
+from tf2_ros.transform_listener import TransformListener
+
+DOCK_ID = int(os.environ.get("SPOT_DOCK_ID", 520))
+FILTER_DIST = 2.4  # in meters (distance for valid detection)
+
+
+class SpotQRDetector:
```

Comment:
Is `SpotQRDetector` defined in multiple places? Why is this another unique class? Does this entail we create `Spot<BespokeDetector>` class for every functionality?

Let's see if we can simplify this. Add detectors to Spot just like loggers accept handlers.

Reply:
For now, I have moved this class to a new file `bd_spot_wrapper/spot_wapper/spot_qr_detector.py`

We can simplify it further in a separate effort as there is an issue already created for this task - https://github.com/facebookresearch/spot-sim2real/issues/90